core/app/[locale]/(default)/cart/_actions/apply-coupon-code.ts

```diff
@@ -0,0 +1,51 @@
+'use server';
+
+import { revalidateTag } from 'next/cache';
+
+import { getSessionCustomerAccessToken } from '~/auth';
+import { client } from '~/client';
+import { graphql, VariablesOf } from '~/client/graphql';
+import { TAGS } from '~/client/tags';
+
+const ApplyCheckoutCouponMutation = graphql(`
+  mutation ApplyCheckoutCouponMutation($applyCheckoutCouponInput: ApplyCheckoutCouponInput!) {
+    checkout {
+      applyCheckoutCoupon(input: $applyCheckoutCouponInput) {
+        checkout {
+          entityId
+        }
+      }
+    }
+  }
+`);
+
+type Variables = VariablesOf<typeof ApplyCheckoutCouponMutation>;
+
+interface Props {
+  checkoutEntityId: Variables['applyCheckoutCouponInput']['checkoutEntityId'];
+  couponCode: Variables['applyCheckoutCouponInput']['data']['couponCode'];
+}
+
+export const applyCouponCode = async ({ checkoutEntityId, couponCode }: Props) => {
+  const customerAccessToken = await getSessionCustomerAccessToken();
+
+  const response = await client.fetch({
+    document: ApplyCheckoutCouponMutation,
+    variables: {
+      applyCheckoutCouponInput: {
+        checkoutEntityId,
+        data: {
+          couponCode,
+        },
+      },
+    },
+    customerAccessToken,
+    fetchOptions: { cache: 'no-store' },
```

Comment:
Not something to address on this PR, but perhaps we should look at `document`'s operation and if it's a mutation we automatically set `fetchOptions.cache` to `no-store`. We never want to cache mutations.

Reply:
Good idea!